receiver/awscontainerinsightreceiver/internal/stores/podstore_test.go

```diff
@@ -261,14 +262,15 @@ func TestPodStore_decorateMem(t *testing.T) {
 	assert.Equal(t, uint64(10*1024*1024), metric.GetField("pod_memory_working_set").(uint64))
 
 	tags = map[string]string{ci.MetricType: ci.TypeContainer, ci.ContainerNamekey: "ubuntu"}
-	fields = map[string]interface{}{ci.MetricName(ci.TypeContainer, ci.MemWorkingset): float64(10 * 1024 * 1024)}
+	fields = map[string]interface{}{ci.MetricName(ci.TypeContainer, ci.MemWorkingset): uint64(10 * 1024 * 1024)}
 
 	metric = generateMetric(fields, tags)
 	podStore.decorateMem(metric, pod)
 
 	assert.Equal(t, uint64(52428800), metric.GetField("container_memory_request").(uint64))
 	assert.Equal(t, uint64(52428800), metric.GetField("container_memory_limit").(uint64))
-	assert.Equal(t, float64(10*1024*1024), metric.GetField("container_memory_working_set").(float64))
+	assert.Equal(t, uint64(10*1024*1024), metric.GetField("container_memory_working_set").(uint64))
```

Comment:
Any reason this changed from a float to a uint, and the new metric is a float?

Reply:
Yea I think this was a mistake in the test. They fixed the same metric at the pod level as part of [open-telemetry/opentelemetry-collector-contrib/pull/3846](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/3846/files#diff-d116179dd6d363b58539c72db351ff53e46f1b4e04ac16565d09a22b564b1b45R261).

The actual code itself treats it as an int as can be seen [here](https://github.com/amazon-contributing/opentelemetry-collector-contrib/blob/aws-cwa-dev/receiver/awscontainerinsightreceiver/internal/cadvisor/extractors/mem_extractor.go#L67).

Why didnt this test fail? Because its being set to a float just above [here](https://github.com/amazon-contributing/opentelemetry-collector-contrib/pull/23/files/906be08432f869842652ca23a73fdd9b60d17d2d#diff-d116179dd6d363b58539c72db351ff53e46f1b4e04ac16565d09a22b564b1b45R265) and its asserting on it being a float here. So this test isnt really going through the mem_extractor code path (which is where type assertion to int64 happens).